### DISCUSSION TITLE
SoapySDR with SDRPlay RSP1A

### DISCUSSION THREAD

**Discussion by thomaswenger** [rtl-airband/RTLSDR-Airband#243] [General]
I have been a LiveATC feeder for our airport for years and have worked a lot with RTL sticks. Basically good, but now I wanted to try something with a larger dynamic range. So I set up a trial run on an Asus Tinker board and the SDRPlay RSP1A.

Basically everything works end-to-end and I could feed like this. However, I am not satisfied with the quality. The signals are much weaker and noisier than with the RTL dongles. I know that RSP1A can do a lot better when I use it with SDRUno - here the signal is just perfect and strong. Not so with SoapySDR - where it is quite weak and noisy.

Since SoapySDR is new territory for me, I suspect suboptimal settings e.g. in the area of gains or similar.

Can anyone give me tips on how to get good results with the SDRPlay device? Have I overlooked something?

Versions:
- Armbian: Linux tinkerboard 5.10.21-rockchip #21.02.3
- SDRplay_RSP_API-ARM32-3.07.2
- SoapySDR 0.8.0
- RTLSDR_Airband 3.2.1

Many thanks & best regards
Tom

[rtl_airband.conf.txt](https://github.com/szpajder/RTLSDR-Airband/files/6411678/rtl_airband.conf.txt)

**Comment by starvald** [comment]
these are the gain options available with the rsp1a:  
gain = "IFGR=34,RFGR=4";
if you try yours with something like cubicsdr which uses soapy to find the optimal settings.

also worth noting the ADC resolution goes down as the bandwidth goes up.
https://www.sdrplay.com/docs/RSP1Adatasheetv1.9.pdf
between 2 and 6mhz will get the best quality. 

14 bit native ADC (2 –6.048MSPS)
12-bit (6.048-8.064 MSPS)
10-bit (8.064-9.216MSPS)
8-bit (> 9.216 MSPS )

**Reply by thomaswenger** [reply]
Thanks! Good hint!

Using separate "gain" values seems to work somehow. One has to be aware of the special concept of "gain" here. It is not gain but gain reduction. Lower values mean higher gain. The best explanation about these settings with SDRPlay devices seems to be this ticket here: https://github.com/pothosware/SoapySDRPlay2/issues/60

Even after having changed these these values a lot I am still not able to get results as good as with RTLSDR. The signal is still a bit on the weak side with some hiss. Which is not the case when running the RSP1A with SDRUno.

**Reply by szpajder** [reply]
I see that you are running at 9 Msps. This means you have to increase fft_size to 1024 or maybe even 2048, because the default value of 512 results in the channel being too wide, so you pick up extra noise.

**Reply by thomaswenger** [reply]
Thanks! My fault. As I normally had the limited sample rate of the RTL sticks I was not used to setting this parameter. One BIG step ahead. That helped a lot.

Now I would say the quality is "quite good". I still have the impression that the audio from an RTL stick is a bit... cleaner...?. And I know that RSP1A can do a lot better e.g. when using with SDRUno.

Thanks to @szpajder and @starvald for your help. Both elements together brought me a huge step nearer to the goal.
